### PR TITLE
change keep ownership when collaborator leaves (or gets kicked out)

### PR DESCRIPTION
**Pull request by ahsu1230**
@andrewconner

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala [hunk 1]
```diff
@@ -1699,32 +1708,39 @@ class LibraryCommander @Inject() (
...
-        case (Some(mem), Some(targetMem)) =>
+        case (Some(mem), Some(targetMem), library) =>
           if ((mem.isOwner && !targetMem.isOwner) || // owners can edit anyone except themselves
             (mem.isCollaborator && mem.userId == targetMem.userId) || // a collaborator can only edit herself
             (mem.isFollower && mem.userId == targetMem.userId)) { // a follower can only edit herself
             db.readWrite { implicit s =>
               newAccess match {
                 case None =>
+                  if (targetMem.access == LibraryAccess.READ_WRITE) {
+                    convertKeepOwnershipToLibraryOwner(targetMem.userId, library)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala]: We probably want to call `convertKeepOwnershipToLibraryOwner` in all cases, for the case of a collaborator who becomes a follower, and then leaves.

**Comment by andrewconner** [comment]
Nice job.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala (hunk 1 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryCommander.scala]: ohh I see. nice catch, thanks!

**Comment by andrewconner** [comment]
Good idea, I remember you mentioning this. @ahsu1230 can you add this when you add the `order` field for ordering keeps? (ie no need to do it immediately, but soon)

**Comment by ahsu1230** [comment]
ok gotcha! sounds good. for `original_keeper_id`, I will backfill them as the current keep's userId

**Pull request merged**